### ISSUE TITLE
`no tags registered` when `compinit` is called by plugin when sourced through `zinit`

### ISSUE THREAD

**Issue by marlonrichert** [zdharma/zinit#366]
**Describe the bug**
When `compinit` is called from within a plugin that is loaded through `zinit light-mode`, then this causes `no tags registered` to be output for the first several times when the user attempts completion (for example, by pressing Tab).

**To Reproduce**
1. Create an empty plugin: `zinit light-mode for zdharma/null`
1. Create a `~/.zinit/plugins/zdharma---null/null.plugin.zsh` and put `autoload -Uz compinit && compinit` in it.
1. In your `.zshrc`, put 
   ```shell
   rm ~/.zcompdump
   zinit light-mode for zdharma/null
   ```
1. Restart your shell.
1. Try a completion.

**Expected behavior**
Completions are shown, just like when I would just `source` the plugin.

**Actual behavior**
Stuff like this:
```shell
_tags:comptags:67: no tags registered
_requested:comptags:8: no tags registered
_next_label:comptags:8: no tags registered
_all_labels:comptags:26: no tags registered
```

**Versions:**
 - $ZSH_VERSION == 5.8
 - Operating system name and version: Darwin Kernel Version 19.5.0; root:xnu-6153.121.2~2/RELEASE_X86_64 x86_64

**Comment by marlonrichert** [comment]
Correction: This output happens only when I also use Powerlevel10k’s instant prompt feature.

**Comment by marlonrichert** [comment]
I found a workaround: Call `.zinit-shade-off $___mode` before calling `compinit` and `.zinit-shade-on $___mode` afterwards.

**Comment by psprint** [comment]
Yes, that would explain it. I'll take a look if something can be done to skip the additional calls.

**Comment by seivan** [comment]
Any update on this? 

So in my case, I let `OMZ` generate the `compdump`, by loading it last. 
and I noticed that I don't get these errors if I generate as well initially, and let `OMZ` overwrite it later, but it keeps trashing around the file. 

```zsh
#zinit
declare -A ZINIT
ZINIT[BIN_DIR]="$ZSH_VENDOR_PATH/zinit"
ZINIT[HOME_DIR]="$ZSH_VENDOR_PATH/zinit_downloads"
ZINIT[ZCOMPDUMP_PATH]="$ZSH_CACHE_DIR"
export ZSH="$ZINIT[HOME_DIR]/plugins/ohmyzsh---ohmyzsh"
source "$ZINIT[BIN_DIR]/zinit.zsh"

#These lines two lines removed, will trigger the issue. 
autoload -Uz compinit
compinit -i -C -d "${ZSH_COMPDUMP}"

source "$ZDOTDIR/zinit_plugins.zsh"
```

As I said, if I remove these two lines
```
autoload -Uz compinit
compinit -i -C -d "${ZSH_COMPDUMP}"
```

I get that error for a few seconds before going away. 


My `zinit_plugins.zsh` look like this 
```zsh
zinit ice lucid wait depth"1"
zinit light zsh-users/zsh-syntax-highlighting

zinit ice lucid wait depth"1"
zinit light zsh-users/zsh-history-substring-search

zinit ice lucid blockf wait depth"1"
zinit light zsh-users/zsh-completions 

plugins=(
z
)
zinit ice lucid nocompletions blockf wait depth"1" 
zinit light ohmyzsh/ohmyzsh

zinit ice lucid nocompletions blockf depth"1"
zinit light denysdovhan/spaceship-prompt
```

**Comment by marlonrichert** [comment]
This popped up again, but in a different way: https://github.com/marlonrichert/zsh-autocomplete/issues/335

@psprint Are you ever going to fix this?

**Comment by marlonrichert** [comment]
Workaround:
```zsh
[[ -v functions[.zinit-shade-off] ]] &&
  .zinit-shade-off "${___mode:-load}"
```
